Validate that `'self'` is present in the init signature of a Parameterized class

maximlt opened the pull request:
Fixes https://github.com/holoviz/param/issues/785

Hoxbro commented on tests/testparameterizedrepr.py:
```diff
@@ -217,3 +218,38 @@ def test_script_repr_parameterized_instance(P):
+            ]
+        )
+
+        with pytest.raises(KeyError, match="'T.__init__.__signature__' must contain a 'self' Parameter."):
```
I would escape `.` with `\.`

Hoxbro reviewed (commented):
I would maybe make a subclass with a custom `__init__` and use that as a base to not mess around with `param.Parameterized` directly.

Maybe also have an example with a parameter in it with a non-default (If it does not already exist.)

This still needs a change in Holoviews, correct?

maximlt commented:
I addressed your comments.

> Maybe also have an example with a parameter in it with a non-default (If it does not already exist.)

`testparameterizedrepr` has a number of `.param.pprint` tests, including testing classes with an `__init__` that has positional only arguments.

Hoxbro commented on tests/testparameterizedrepr.py:
```diff
@@ -217,3 +218,39 @@ def test_script_repr_parameterized_instance(P):
+
+    t = T()
+
+    # This is actually setting the signature of param.Parameterized.__init__
+    # as P doesn't define __init__
```
```suggestion
    # This is actually setting the signature of P.__init__
    # as T doesn't define __init__
```
